chainer/functions/loss/sigmoid_cross_entropy.py

```diff
@@ -30,50 +25,39 @@ def forward(self, inputs):
         xp = cuda.get_array_module(*inputs)
         x, t = inputs
         self.ignore_mask = (t != self.ignore_label)
-        if self.normalize:
-            count = xp.maximum(1, self.ignore_mask.sum())
-        else:
-            count = max(1, len(x))
-        self.count = count
         # stable computation of the cross entropy.
-        loss = -xp.sum(
-            self.ignore_mask * (x * (t - (x >= 0)) -
-                                xp.log1p(xp.exp(-xp.abs(x)))))
-        return utils.force_array(xp.divide(loss, self.count, dtype=x.dtype)),
+        loss = -(
+            self.ignore_mask *
+            (x * (t - (x >= 0)) - xp.log1p(xp.exp(-xp.abs(x)))))
+
+        return utils.force_array(loss.astype(x.dtype)),
 
     def backward(self, inputs, grad_outputs):
-        xp = cuda.get_array_module(*inputs)
         x, t = inputs
         gloss = grad_outputs[0]
         y, = sigmoid.Sigmoid().forward((x,))
-        gx = xp.divide(
-            gloss * self.ignore_mask * (y - t), self.count,
-            dtype=y.dtype)
+        gx = (gloss * self.ignore_mask * (y - t)).astype(y.dtype)
         return gx, None
 
 
-def sigmoid_cross_entropy(x, t, normalize=True):
+def sigmoid_cross_entropy(x, t):
     """Computes cross entropy loss for pre-sigmoid activations.
 
     Args:
-        x (Variable): A variable object holding a matrix whose (i, j)-th
-            element indicates the unnormalized log probability of the j-th unit
-            at the i-th example.
-        t (Variable): Variable holding an int32 vector of ground truth labels.
-            If ``t[i] == -1``, corresponding ``x[i]`` is ignored.
-            Loss is zero if all ground truth labels are ``-1``.
-        normalize (bool): Variable holding a boolean value which
-            determines the normalization constant. If true, this function
-            normalizes the cross entropy loss across all instances. If else,
-            it only normalizes along a batch size.
+        x (Variable): Variable holding a float32 array whose elements
+            indicate the unnormalized log probabilities.
+        t (Variable): Variable holding an int32 array of ground truth labels.
+            If ``t[i0, i1, ... ] == -1``, corresponding ``x[i0, i1, ... ]`` is
+            ignored. In this case, the loss value of the ignored instance is
+            set to ``0``.
```

Comment:
Do you think we should accept a scalar-input case (i.e. `x.shape` and `t.shape` are `()`)? In that case, `utils.force_array` could be needed.